yt_idefix/data_structures.py

```diff
@@ -427,15 +427,45 @@ def _get_time(self):
 
     def _set_code_unit_attributes(self):
         """Conversion between physical units and code units."""
+        # Get units defined in definitions.h
         length_unit = self.parameters.get("unit_length", 1.0)
         velocity_unit = self.parameters.get("unit_velocity", 1.0)
         density_unit = self.parameters.get("unit_density", 1.0)
 
-        self.length_unit = self.quan(length_unit, "cm")
-        self.velocity_unit = self.quan(velocity_unit, "cm/s")
-        density_unit = self.quan(density_unit, "g/cm**3")
+        # Get override_units
+        uo = self.units_override
+        self.length_unit = uo.get("length_unit", self.quan(length_unit, "cm"))
+        self.velocity_unit = uo.get("velocity_unit", self.quan(velocity_unit, "cm/s"))
+        density_unit = uo.get("density_unit", self.quan(density_unit, "g/cm**3"))
 
         self.mass_unit = density_unit * self.length_unit ** 3
         self.time_unit = self.length_unit / self.velocity_unit
         self.magnetic_unit = np.sqrt(4.0 * np.pi * density_unit) * self.velocity_unit
         self.magnetic_unit.convert_to_units("gauss")
+
+    default_units = {
+        "length_unit": "cm",
+        "time_unit": "s",
+        "density_unit": "g/cm**3",
```

Comment:
please add a comment here to note that `"density_unit"` is the one key that's actually not in the base `Dataset` class